### ISSUE TITLE
How to use auto-sklearn with BOHB?

### ISSUE THREAD

**Issue by alexgcsa** [automl/auto-sklearn#654]
Hello,

I was watching the last AutoML tutorial (from NeuIPs with Frank Hutter and Joaquin Joaquin Vanschoren), and, in the first part, it is said that it is possible to use Auto-sklearn with BOHB (i.e., the hybrid method that combines the benefits of Hyperband and Bayesian Optimization).

[Slide 37 of the NeurIPS presentation](https://www.automl.org/wp-content/uploads/2018/12/AutoML-Tutorial-NeurIPS2018-HPO_and_NAS.pdf).

I was wondering how I can use this feature on Auto-sklearn. I have tried to search on the API section but I have not found it. 

Cheers,

Alex de Sá

**Comment by mfeurer** [comment]
Hi,

The combination of Auto-sklearn with successive halving/hyperband/BOHB is so far only available as a competition submission from the 2nd AutoML competition: http://ml.informatik.uni-freiburg.de/downloads/automl_competition_2018.zip

A full integration of hyperband into Auto-sklearn is on our todo list, but there's no timeline on when we'll do this.

You can find a description here: https://ml.informatik.uni-freiburg.de/papers/18-AUTOML-AutoChallenge.pdf

Cheers,
Matthias

**Comment by alexgcsa** [comment]
Thanks for the answer, Matthias.

Cheers,

Alex

**Comment by alexgcsa** [comment]
Hi,

could you give one example of the input?

In the run.py, it is said that:

**The input directory input_dir contains 5 subdirectories named by dataset, including:**

- dataname/dataname_feat.type          -- the feature type "Numerical", "Binary", or "Categorical" (Note: if this file is abscent, get the feature type from the dataname.info file)
- dataname/dataname_public.info        -- parameters of the data and task, including metric and time_budget
- dataname/dataname_test.data          -- training, validation and test data (solutions/target values are given for training data only)
- dataname/dataname_train.data
- dataname/dataname_train.solution
- dataname/dataname_valid.data

But I would like to know how these files are formatted (mainly the first two files).

Cheers,

Alex

**Comment by mfeurer** [comment]
The file formatting is according to the data format of the 2nd AutoML challenge. You can find a description and example datasets [here](https://competitions.codalab.org/competitions/17767#participate).
